terraform/modules/happy-service-ecs/main.tf

```diff
@@ -24,6 +24,111 @@ resource "aws_ecs_service" "service" {
 
 locals {
   task_definition = [
+    {
+      name      = "datadog-agent"
+      essential = true
+      image     = "public.ecr.aws/datadog/agent:latest"
+      cpu       = 512
+      memory    = 512
+
+      environment = concat(
+        [
+          {
+            name  = "DD_API_KEY"
+            value = var.datadog_api_key
```

Comment:
this is probably okay for now, but I wonder if instead of value, we try to use something like this:

~~~
{
  "containerDefinitions": [{
    "secrets": [{
      "name": "environment_variable_name",
      "valueFrom": "arn:aws:secretsmanager:region:aws_account_id:secret:appauthexample-AbCdEf:username1::"
    }]
  }]
}
~~~

Reply:
https://aws.amazon.com/premiumsupport/knowledge-center/ecs-data-security-container-task/

Reply:
I don't think we should do it at the service level. Feeding it into the integration secret this way, perhaps.